File: cpp/unittests/FindMedianFromDataStreamTest.cpp
#include "FindMedianFromDataStream.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(FindMedianFromDataStream, Solution295_1) {
  MedianFinder1 mf;
  mf.addNum(1);
  mf.addNum(2);
  EXPECT_EQ(1.5, mf.findMedian());
  mf.addNum(3);
  EXPECT_EQ(2, mf.findMedian());
}

File: java/src/test/java/io/binac/leetcode/FindMedianFromDataStreamTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class FindMedianFromDataStreamTest {

    @Test
    void test1() {
        FindMedianFromDataStream.MedianFinder medianFinder = new FindMedianFromDataStream.MedianFinder();
        medianFinder.addNum(1);
        medianFinder.addNum(2);
        assertEquals(1.5, medianFinder.findMedian());
        medianFinder.addNum(3);
        assertEquals(2, medianFinder.findMedian());
    }
}